src/shelf_usage/zone.cc

```diff
@@ -554,6 +556,13 @@ void Zone::free(Offset block) {
     zoneheader->free_list[level].push(header_ptr, block/nvmm_read(&zoneheader->min_obj_size));
 }
 
+void Zone::clear_data(Offset block) {
```

Comment:
i think you should call clear_data function from zone->Free itself if fast alloc is set. (do not need to call clear_data separately from delayed_free_fn)